### ISSUE TITLE
Code cells are being spellchecked

### ISSUE THREAD

**Issue by Nicholaswogan** [jupyterlab-contrib/spellchecker#141]
In many cases, this extension is checking the spelling in "code" cells. Is this expected behavior? Can this behavior be turned off?

![Screenshot 2024-01-22 at 1 52 49 PM](https://github.com/jupyterlab-contrib/spellchecker/assets/18489667/55a0d3ab-e6f5-40f3-9c69-dde71db462c5)

**Comment by krassowski** [comment]
Yes and no. By default only comments are checked in code, which is opt-out. String checking are opt-in. What do you see in your settings (go to menu bar Settings → Settings Editor → Spellchecker); this should look like:

![image](https://github.com/jupyterlab-contrib/spellchecker/assets/5832902/aeb8fccd-856a-4bfd-9739-dbb016416fec)

**Comment by Nicholaswogan** [comment]
This is what my settings looks like:

![Screenshot 2024-01-23 at 8 13 40 AM](https://github.com/jupyterlab-contrib/spellchecker/assets/18489667/83db8dd1-eb24-4a86-bd84-fc8c2e9c490a)

**Comment by bradezard131** [comment]
Also seeing this issue, getting misspelling notes on imports, class names, etc. and no obvious way to turn it off
![image](https://github.com/jupyterlab-contrib/spellchecker/assets/2363042/e7e906bc-a1d4-45f8-84cb-040f7f019ca5)

**Comment by krassowski** [comment]
@bradezard131 is this in a notebook or in the file editor? What browser are you using? If you type something in the editor does it go away?

**Comment by bradezard131** [comment]
This is a notebook running in chrome. It does appear to go away when I start typing in each cell, which I'll use as a workaround for now to clean it up. I'd hope that this is still considered a bug though that I have to type in every cell to clean up the attention-grabbing red boxes.

**Comment by krassowski** [comment]
> It does appear to go away when I start typing in each cell, which I'll use as a workaround for now to clean it up

Ok, I see what is happening. It has nothing to do with the feature of spelling in code. Instead, the notebook initially advertises the cell to contain text and only switches the MIME format to Python code after the kernel has connected. Thus, the extension thinks that it is checking plain text. I never seen this in the notebook before but I did see this issue in File editor.

The solution would be to ensure we refresh the spellchecking after the editor MIME type changed.

**Comment by shivendra90** [comment]
Happening here as well. A good workaround is to close the notebook from the tab (don't shut down the kernel) and open it again. Those annoying highlights will go away.

**Comment by krassowski** [comment]
> Ok, I see what is happening. It has nothing to do with the feature of spelling in code. Instead, the notebook initially advertises the cell to contain text and only switches the MIME format to Python code after the kernel has connected. Thus, the extension thinks that it is checking plain text. 

Someone just shared this behaviour with me for a slow-starting kernel which did not report its code language until a dozen seconds or so after the notebook loaded.
